### ISSUE TITLE
`Problems reading data from Binary store` error

### ISSUE THREAD

**Issue by emartynov** [gradle/gradle#8489]
After update to Gradle 5.2 we started noticing build failures with missed dependencies on the disk. For example:
```
> Could not resolve all dependencies for configuration ':app:acceptanceRuntimeClasspath'.
   > Problems reading data from Binary store in /private/var/folders/j9/m08k8g8d6_d1qqp714rqbhgr0000gv/T/gradle908977521929543055.bin (exist: false)
```
### Context
We have a multi-module android project with a mix of Kotlin library, Android library and Android application modules.
Sometimes similar error happens with kapt task.

We use Gradle 5.2, Kotlin 1.3.21 and Android Gradle Plugin 3.4-beta-03. We use Gradle cache server that is hosted in local network.

This is the content of our gradle.properties:
```
org.gradle.jvmargs=-Xmx4g -XX:MaxMetaspaceSize=2g -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8

# When configured, Gradle will run in incubating parallel mode.
# This option should only be used with decoupled projects. More details, visit
# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
org.gradle.parallel=true

# Configure Gradle build cache.
# https://docs.gradle.org/current/userguide/build_cache.html
# Disbaled while checking if build is still failing
org.gradle.caching=true

# Enable experimental feature to allow Gradle to configure only necessary projects.
# https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:configuration_on_demand
org.gradle.configureondemand=true

# Experimental flag to allow Kapt to use Gradle's Worker API.
# Disabled because it causes OutOfMemoreyExceptions in combination with Android Gradle Plugin 3.4 beta.
kapt.use.worker.api=false

# Disable the new code shrinking/obfuscation tool R8, we use DexGuard instead.
android.enableR8=false

# Enable AndroidX
android.useAndroidX=true
android.enableJetifier=true

# Required for Robolectric 4.0
android.enableUnitTestBinaryResources=true
```
There are gradle additional properties that we set on CI:
```
-Dorg.gradle.daemon=false -Dorg.gradle.parallel=true -Dorg.gradle.workers.max=2 -Dkotlin.incremental=false
```

### Steps to Reproduce (for bugs)
Unfortunately, I can not provide own project as well it only fails for specific feature branch. Rest looks unaffected.

### Your Environment
It fails and on Mac and Linux build agents. We use gitlab runners to run our CI. I can not reproduce same on the local machine.

**Comment by emartynov** [comment]
Any direction I need to take, trying to solve it?

**Comment by melix** [comment]
Thanks for the report. The fact the file is _missing_ is very fishy. I see you are using both configure on demand and parallel. Can you try to see if disabling one or both is fixing the problem?

**Comment by emartynov** [comment]
Disabling the configuration on demand didn't help :(

**Comment by emartynov** [comment]
Looks like parallel disabling also didn't help.

This my current test gradle configuration line:
```
  GRADLE_OPTS: "-Dorg.gradle.daemon=false -Dorg.gradle.parallel=true -Dorg.gradle.workers.max=2 -Dkotlin.incremental=false -Dorg.gradle.configureondemand=false -Dorg.gradle.parallel=false"
```

What has higher prio - env variables or properties file in the project?

**Comment by emartynov** [comment]
Oke from here (https://docs.gradle.org/current/userguide/build_environment.html) I'm doing it worng.

Let me retest it.

**Comment by emartynov** [comment]
Finished tests and it still fails.

```
Execution failed for task ':libs-rx:kaptGenerateStubsTestKotlin'.
> Could not resolve all dependencies for configuration ':libs-rx:testCompileClasspath'.
   > Problems reading data from Binary store in /private/var/folders/4g/b0r_2ml50z58789ck8hx8p7c0000gs/T/gradle4325000891140356809.bin (exist: false)
```

It is in a different place now. But I can not say it is because of mac agent or changes that I just made.

**Comment by emartynov** [comment]
Good morning! Any other ideas?

**Comment by melix** [comment]
A build scan would help. I'm interested in knowing if you have any warning in the logs. I'm wondering if this is something about GitLab, since the location is unusual.

**Comment by emartynov** [comment]
I have it but don't want to paste public. How could I share it?

**Comment by emartynov** [comment]
Oke, looking to scan I see that `parallel` was still on. So after I disabled it the build passed.

**Comment by emartynov** [comment]
So from different runs, I don't have a clear understanding of what is the cause. I got it once successfully finished and all the rest tries failed.

**Comment by melix** [comment]
So when you say the "rest tried failed", you mean _without_ parallel? If parallel is on I suspect 2 different threads are trying to read the same resolution result at the same time, but somehow defeat the protection we have around this. However protection is only supposed to work if it's Gradle managed threads so it would really help if you managed to have a reproducible example.

**Comment by emartynov** [comment]
I've sent you scans by email. Let me know if you need more information.

**Comment by big-guy** [comment]
@emartynov Which version of Gradle did you upgrade from?

**Comment by emartynov** [comment]
We've been on 4.10.3 and then moved to 5.2. And recently to 5.2.1.

**Comment by melix** [comment]
So looking at the build scans you sent privately, I see no explanation so far. The code which reports the file is missing should never reach that state.

One possibility is that the CI server uses a weird file system which doesn't flush when we ask it to. But even with this, it would not explain a difference between 4.10.3 and 5.2.

I would be interested in the result of running with `--debug`. The log is going to be huge, so do not hesitate to send it to me zipped by email.

**Comment by emartynov** [comment]
174 Mb of log file zipped to 7.5 are sent to you. Please let me know if I can do something else.

**Comment by lalilulelo1986** [comment]
I have similar problem

`* What went wrong:
Execution failed for task ':admin-console:compileKotlin'.
> Could not resolve all dependencies for configuration ':admin-console:detachedConfiguration48'.
   > Problems reading data from Binary store in /tmp/gradle7646993990205285778.bin (exist: false)`

**Comment by lalilulelo1986** [comment]
`
* What went wrong:
Execution failed for task ':admin-console:compileKotlin'.
\> Could not resolve all dependencies for configuration ':admin-console:detachedConfiguration48'.
   \> Problems reading data from Binary store in /tmp/gradle7646993990205285778.bin (exist: false)`

> I have similar problem:
> 
> * What went wrong:
> Execution failed for task ':admin-console:compileKotlin'.
> > Could not resolve all dependencies for configuration ':admin-console:detachedConfiguration48'.
> > Problems reading data from Binary store in /tmp/gradle7646993990205285778.bin (exist: false)

It seems like 2 different pipelines conflicts with each other and deletes each other data temp files

**Comment by greyskyy** [comment]
I saw this error also, but i was able to resolve it.  My issue was caused by pulling in a jdk11 jar into a build with source-compatibility=1.8

not sure if that'll fix your issue or not, but maybe double check your java version against your dependencies?

**Comment by ljacomet** [comment]
Closing as this could not be reproduced.

**Comment by archmisha** [comment]
I had the same issue on gradle 6.0.1.
Upgrading to 6.2 fixed the issue
(upgrading to 6.1.1 didn't fix the issue, running with --no-parallel on 6.0.1 also fixed the issue)

**Comment by abdullahizzuddiin** [comment]
I got this issue too, even I have used gradle v6.2.

This error doesn't occurred when I ran `./gradlew build` after I cleaned gradle `./gradlew clean`

**Comment by nniesen** [comment]
The problem is that is completely random. I had a build fail 3 times on CI server and then 3 hours later, it ran just fine with no changes. The same error couldn't be reproduced locally at the same time. The issue is present in 6.5 and all the way back to at least 3.2, from search. It seems to be a threading issue where Gradle deletes it's own temp file before it should. That seems to be why random changes "fix" the issue. It would be helpful if the error reported the dependency in question but that might not help since the entire temp file is not present. 

From a recent Gradle 6.5 build:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateMetadataFileForMavenPublication'.
> Could not resolve all dependencies for configuration ':runtimeClasspath'.
   > Problems reading data from Binary store in /tmp/gradle3396894650886991501.bin offset 71292 exists? true
```

**Comment by MauricioGenerosoTR** [comment]
@nniesen I Have the same error in a project using 6.4

```
 What went wrong:
Problems reading data from Binary store in C:\Users\XXXX\AppData\Local\Temp\gradle16994281476084561545.bin offset 958194 exists? true
```

**Comment by ljacomet** [comment]
If you can provide a reproducer, please open a new issue. Otherwise, we have and are still fixing these issues when we can reproduce them.
Try with Gradle 6.5 or even a [6.6 nightly](https://gradle.org/nightly/)

**Comment by nniesen** [comment]
I don't have something reproducible but haven't distilled it down to something that I can share yet so just adding info here for anyone encountering my last post. I can't verify that all the similar errors we've recently encounter are attributed to this but I think the majority of them are stemming from our switch to using dynamic versions for some dependencies. There are a few scenarios but one that is reproducible and diagnosible is projects not producing .pom or .module files with resolved versions by including a versionMapping block in the publish. When downstream projects, in this particular case a 'platform' project, try to resolve versions for the dependencies or publishing tasks they generate failures like the following. In this case, the stacktrace at least identifies the artifact causing the error even if it's not clear what's wrong.

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':dependencies'.
> Could not resolve all dependencies for configuration ':platformApiClasspath'.
   > Problems reading data from Binary store in C:\Users\nniesen\AppData\Local\Temp\gradle4965728309246164743.bin offset 9762 exists? true

[...]

Caused by: java.lang.IllegalStateException: Corrupt serialized resolution result. Cannot find selected module (153) for platformApiClasspath -> com.xxx.yyy:azure-client
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.DefaultResolutionResultBuilder.visitOutgoingEdges(DefaultResolutionResultBuilder.java:82)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.result.StreamingResolutionResultBuilder$RootFactory.deserialize(StreamingResolutionResultBuilder.java:237)
        at org.gradle.api.internal.artifacts.ivyservice.resolveengine.store.DefaultBinaryStore$SimpleBinaryData.read(DefaultBinaryStore.java:130)
```

In the above case, the azure-client dependency had published .pom and .module files with dependencies that still had dynamic (+) patch versions including our platform (bom). Trying to execute the dependencies or publish tasks for our platform, which in turn has a dynamic version of azure-client, fails with the above error. Our platform build has dynamic dependencies that are fully resolved before publishing (https://github.com/gradle/gradle/issues/8887).

**Comment by pedrolamarao** [comment]
In my case, this error manifested as a consequence of our artifact repository service being down. After restoring it up, the error was gone.

**Comment by nniesen** [comment]
I really appreciate all the work has gone into 6.5-6.7 to stabilize and make dependency resolution more consistent. Thank you!

**Comment by marchacio** [comment]
I solved upgrading:

**_project app-level build.gradle file:_**
com.google.firebase:firebase-analytics:17.2.2    to   com.google.firebase:firebase-analytics:18.0.2

**_Project-level build.gradle file:_**
com.google.gms:google-services:4.3.3    to    com.google.gms:google-services:4.3.5

**Comment by iamshubhamsaroj** [comment]
> I solved upgrading following packages to last version (06/02/2021), so:
> 
> **_project app-level build.gradle file:_**
> com.google.firebase:firebase-analytics:17.2.2 to com.google.firebase:firebase-analytics:18.0.2
> 
> **_Project-level build.gradle file:_**
> com.google.gms:google-services:4.3.3 to com.google.gms:google-services:4.3.5

**Comment by iamshubhamsaroj** [comment]
i was trying to solve this since 2 days  but no one of them worked .
Thanks marchacio😃😃

**Comment by mohmedelsayed** [comment]
i was solved it today by writing 'temp' in 'windows Run commands '  and deleting temp folder .. also '%temp%' and also deleted
😃😃also i was delate project Build folder and i was rebuild again then it solved

**Comment by yeeemayy** [comment]
> I solved upgrading following packages to last version (06/02/2021), so:
> 
> **_project app-level build.gradle file:_**
> com.google.firebase:firebase-analytics:17.2.2 to com.google.firebase:firebase-analytics:18.0.2
> 
> **_Project-level build.gradle file:_**
> com.google.gms:google-services:4.3.3 to com.google.gms:google-services:4.3.5

In my case I followed @marchacio 's way with a few extra steps for the issue to be solved.
Extra steps:
1. delete 'pubspec.lock'
2. Run 'flutter clean'
3. run the project again

**Comment by alex-mironov** [comment]
@marchacio can you please suggest where can we check the relevant versions of com.google.firebase:firebase-analytics and com.google.gms:google-services? I assume they've changed since your post, but I cannot really understand where to look for them. Thanks in advance!

**Comment by marchacio** [comment]
> @marchacio can you please suggest where can we check the relevant versions of com.google.firebase:firebase-analytics and com.google.gms:google-services? I assume they've changed since your post, but I cannot really understand where to look for them. Thanks in advance!

Hi @alex-mironov , the versions I wrote previously may be old and upgradeable.  Apparently they seem to be some of the few versions without bugs😂 however I found this link which lists all the updated versions, hopefully it helps.

https://firebase.google.com/support/release-notes/android#latest_sdk_versions

**Comment by pktippa** [comment]
and if you are using `react-native-push-notification` make sure you update to `7.3.1`, check here https://github.com/zo0r/react-native-push-notification/issues/1999#issuecomment-840235222

**Comment by jack7891** [comment]
I've resolved mine by updating these files:

android/build.gradle
`classpath 'com.google.gms:google-services:4.3.4'` to `classpath 'com.google.gms:google-services:4.3.5'`

android/app/build.gradle
`implementation 'com.google.firebase:firebase-analytics:17.3.0'` to `implementation 'com.google.firebase:firebase-analytics:18.0.2'`

package.json
`    "@react-native-firebase/analytics": "^11.1.2",
    "@react-native-firebase/app": "^11.1.2",
    "@react-native-firebase/auth": "^11.1.2",
    "@react-native-firebase/crashlytics": "^11.1.2",
    "react-native-push-notification": "^7.2.1",
`
to 

`    "@react-native-firebase/analytics": "^11.5.0",
    "@react-native-firebase/app": "^11.5.0",
    "@react-native-firebase/auth": "^11.5.0",
    "@react-native-firebase/crashlytics": "^11.5.0",
    "react-native-push-notification": "^7.3.1",
`

Then, execute npm install to update the modules.
Enter android folder and ./gradlew.bat clean
At root, rebuild the app. (Did the clean and rebuild twice to get it working).

Hope it helps.

**Comment by muffu-7** [comment]
> I've resolved mine by updating these files:
> 
> android/build.gradle
> `classpath 'com.google.gms:google-services:4.3.4'` to `classpath 'com.google.gms:google-services:4.3.5'`
> 
> android/app/build.gradle
> `implementation 'com.google.firebase:firebase-analytics:17.3.0'` to `implementation 'com.google.firebase:firebase-analytics:18.0.2'`
> 
> package.json
> 
> ```
> "@react-native-firebase/analytics": "^11.1.2",
> "@react-native-firebase/app": "^11.1.2",
> "@react-native-firebase/auth": "^11.1.2",
> "@react-native-firebase/crashlytics": "^11.1.2",
> "react-native-push-notification": "^7.2.1",
> ```
> 
> to
> 
> ```
> "@react-native-firebase/analytics": "^11.5.0",
> "@react-native-firebase/app": "^11.5.0",
> "@react-native-firebase/auth": "^11.5.0",
> "@react-native-firebase/crashlytics": "^11.5.0",
> "react-native-push-notification": "^7.3.1",
> ```
> 
> Then, execute npm install to update the modules.
> Enter android folder and ./gradlew.bat clean
> At root, rebuild the app. (Did the clean and rebuild twice to get it working).
> 
> Hope it helps.

Thanks a lot @jack7891 this solution solution woked for me. Although i did not have to clean and rebuild twice

**Comment by Thilanka-97** [comment]
> and if you are using `react-native-push-notification` make sure you update to `7.3.1`, check here [zo0r/react-native-push-notification#1999 (comment)](https://github.com/zo0r/react-native-push-notification/issues/1999#issuecomment-840235222)

Thank you so much.

**Comment by bpsourav21** [comment]
```
* What went wrong:
Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.
> Could not resolve all dependencies for configuration ':app:debugCompileClasspath'.
   > Problems reading data from Binary store in C:\Users\sourav\AppData\Local\Temp\gradle7627481342141045842.bin offset 125802 exists? true
```

I am getting this error from yesterday and It is barely building my app. I am using `gradle-6.2-all.zip` 

here is my `..\android\build.gradle` file

```
buildscript {
    ext {
        buildToolsVersion = "29.0.3"
        minSdkVersion = 21
        compileSdkVersion = 29
        targetSdkVersion = 29

        googlePlayServicesIidVersion = "16.0.1"
        firebaseMessagingVersion = "21.1.0"
        playServicesVersion = "17.0.0" 
        androidMapsUtilsVersion = "2.2.1"
    }
    repositories {
        google()
        jcenter()
        mavenCentral()
    }
    dependencies {
        classpath("com.android.tools.build:gradle:3.5.4")
        classpath('com.google.gms:google-services:4.3.3')
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            url("$rootDir/../node_modules/react-native/android")
        }
        maven {
            url("$rootDir/../node_modules/jsc-android/dist")
        }

        google()
        jcenter()
        maven { url 'https://www.jitpack.io' }
    }
}
```

Anyone can suggest how can I fix my build issue and what might be the problem here?

**Comment by mannnish** [comment]
### FOR FLUTTER
https://github.com/flutter/flutter/issues/85972

**Comment by ljacomet** [comment]
@bpsourav21 These issues are generally a bug in the Gradle dependency resolution engine. A number of them were fixed during the 6.x line. So first recommendation would be to upgrade to Gradle 6.9 and see how that goes.

**Comment by hussainsherwani** [comment]
@ljacomet  still facing the same issue in Gradle 6.9 . for me before it was working fine today I am getting this issue while I didn't change anything this

**Comment by bpsourav21** [comment]
Thanks, @ljacomet for your suggestion. Actually, after some intense debugging, I found the reason was for `react-native-braintree-dropin-ui ` this library which I am using on my app. Below is the detailed information about that issue -
wgltony/react-native-braintree-dropin-ui#75

I forked that original project and made some changes for making my app working again. Sharing that fork project below, if anyone needs help -
https://github.com/bpsourav21/react-native-braintree-dropin-ui

In short, I didn't require updating my Gradle version. Only updating that library changes my app again started working and haven't encountered such issue further.

**Comment by dharshu5** [comment]
> Thanks, @ljacomet for your suggestion. Actually, after some intense debugging, I found the reason was for `react-native-braintree-dropin-ui ` this library which I am using on my app. Below is the detailed information about that issue -
> [wgltony/react-native-braintree-dropin-ui#75](https://github.com/wgltony/react-native-braintree-dropin-ui/issues/75)
> 
> I forked that original project and made some changes for making my app working again. Sharing that fork project below, if anyone needs help -
> https://github.com/bpsourav21/react-native-braintree-dropin-ui
> 
> In short, I didn't require updating my Gradle version. Only updating that library changes my app again started working and haven't encountered such issue further.

Which Lib you have updated?...because in my case i didn't used this plugin react-native-braintree-dropin-ui

**Comment by yahorsi** [comment]
I have Gradle 6.9.1 and still have this issue ((

**Comment by tbarusseau** [comment]
I just had this issue happen on a project where nothing changed since a few days and I've been able to build several times without issues.

Now it just fails before even starting to build the app:

> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Could not determine the dependencies of task ':app:transformClassesAndResourcesWithR8ForRelease'.
> > Failed to notify dependency resolution listener.
>    > Problems reading data from Binary store in /tmp/gradle3595031159872134751.bin offset 103873 exists? true
>    > Problems reading data from Binary store in /tmp/gradle3595031159872134751.bin offset 103873 exists? true

If I watch the `/tmp/` directory during the `gradlew` execution, I can see this file being created, and then deleted around the time the build fails.

Since the general consensus seems to be that it's pretty much random, I'll try upgrading a few dependencies and maybe move things around in the Android code.

**Comment by pt2121** [comment]
I am running into this issue today too. The project had been built fine until today. I don't recall making any changes to any build files recently. Somehow, It just stop building when I ran the project this morning and failed to build since then. Very interesting :thinking: 
The project is React Native and I tried solutions which were mentioned in this thread.
Any other ideas?

**Comment by tbarusseau** [comment]
Could it come from the fact that jcenter.bintray.com is down? And the error isn't reported correctly? It's honestly very far-fetched but it's disrupting a lot of builds right now.

**Comment by skeie** [comment]
Exact same thing happend for us @pt2121. For me the error showed the package that was causing it. 

For us it was `implementation 'com.facebook.android:facebook-android-sdk:[5,6)`

I changed it to `implementation 'com.facebook.android:facebook-android-sdk:6` and then it worked fine

**Comment by pt2121** [comment]
@tbarusseau I'm unsure if it's related to jcenter.bintray.com, maybe? I still can't find a solution
@skeie glad that worked out for you. For me, we don't use the dependency.

**Comment by lchwe** [comment]
replace jcenter() with mavenCentral() in the android/build.gradle file, then flutter clean

**Comment by tbarusseau** [comment]
I can confirm that in my case, it was the downtime from jcenter.bintray.com that caused this "unreproducible" issue. So it seems that the file deletion is caused by something else in the first place. This morning, I haven´t touched anything, but the build works once again.

Also @lchwe is right since the service is supposed to go down definitely at some point.

**Comment by pt2121** [comment]
Same here. My build runs fine after JCenter got back up online. We will need to migrate to Maven Central too. Thanks all.

**Comment by pedrolamarao** [comment]
Does this imply that this problem will happen if Maven Central gets offline, heaven forbid?

**Comment by sabin10** [comment]
Something that fixed for me:

<img width="1315" alt="Screenshot 2022-05-10 at 11 29 58" src="https://user-images.githubusercontent.com/35928720/167584903-c2e42db4-8261-4310-bfa0-fdac378b493d.png">

**Comment by sameersitre** [comment]
> Exact same thing happend for us @pt2121. For me the error showed the package that was causing it.
> 
> For us it was `implementation 'com.facebook.android:facebook-android-sdk:[5,6)`
> 
> I changed it to `implementation 'com.facebook.android:facebook-android-sdk:6` and then it worked fine

thanks it worked.

**Comment by zhenguet** [comment]
`I've resolved mine by updating these files:

android/build.gradle
classpath 'com.google.gms:google-services:4.3.4' to classpath 'com.google.gms:google-services:4.3.5'

android/app/build.gradle
implementation 'com.google.firebase:firebase-analytics:17.3.0' to implementation 'com.google.firebase:firebase-analytics:18.0.2'

package.json

"@react-native-firebase/analytics": "^11.1.2",
"@react-native-firebase/app": "^11.1.2",
"@react-native-firebase/auth": "^11.1.2",
"@react-native-firebase/crashlytics": "^11.1.2",
"react-native-push-notification": "^7.2.1",
to

"@react-native-firebase/analytics": "^11.5.0",
"@react-native-firebase/app": "^11.5.0",
"@react-native-firebase/auth": "^11.5.0",
"@react-native-firebase/crashlytics": "^11.5.0",
"react-native-push-notification": "^7.3.1",
Then, execute npm install to update the modules.
Enter android folder and ./gradlew.bat clean
At root, rebuild the app. (Did the clean and rebuild twice to get it working).

Hope it helps.`

thank you @jack7891 .
With me. I just update 'com.google.firebase:firebase-analytics:17.3.0' to 'com.google.firebase:firebase-analytics:18.0.1'

**Comment by HJunLong601** [comment]
if your project do not work in new env,try different version AS

**Comment by nniesen** [comment]
Run your build with `--stacktrace`. The stacktrace messaging is pretty good now at indicating which dependency is causing issues for dependency resolution. Usually the cause I see is some dependency with a broken reference or one containing an unresolved version that's causing a cycle.

**Comment by abdelrahmanelmarakby** [comment]
On Windows:

```gradlew cleanBuildCache```
On Mac or UNIX:

```./gradlew cleanBuildCache```

That solution fixed the error for me

**Comment by yigit** [comment]
What is interesting is that, it is failing while trying to access 2.6.0
```
Caused by: java.lang.IllegalStateException: Corrupt serialized resolution result. Cannot find selected module (182) for releaseVariantReleaseRuntimePublication -> androidx.lifecycle:lifecycle-viewmodel-savedstate:2.6.0
```
while the project [depends on 2.6.1](https://ge.androidx.dev/s/cmk3rsfzjiacc/dependencies?toggled=W1sxMV0sWzExLDBdLFsxMSwwLFsxODE2XV0sWzExLDAsWzE4MTNdXV0)

And there is no dependency on 2.6.0 in the source code.

We also use dependency constraints so maybe it is related in some way?

**Comment by GokhanArik** [comment]
@yigit Similar scenario. 

This happened a while back and I updated it from `2.5.1` to `2.6.1`, and the problem disappeared.

Now this is happening again but this time we don't even depend on `2.5.1`.

```
Corrupt serialized resolution result. Cannot find selected module (1122) for releaseVariantReleaseApiPublication -> androidx.lifecycle:lifecycle-livedata-core-ktx:2.5.1
```

**Comment by nniesen** [comment]
@GokhanArik It has little or nothing to do with the Gradle version. It varies based the order the dependencies are processed which can change for a number of reasons. See my [comment](https://github.com/gradle/gradle/issues/8489#issuecomment-1205383606) about using `--stacktrace` to identify the dependency likely causing the issue.
